src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -3430,10 +3430,15 @@ bool IdealLoopTree::iteration_split_impl(PhaseIdealLoop *phase, Node_List &old_n
         return false;
       }
 
-      // We are going to add pre-loop and post-loop.
-      // But should we also multi-version for auto-vectorization speculative
-      // checks, i.e. fast and slow-paths?
-      phase->maybe_multiversion_for_auto_vectorization_runtime_checks(this, old_new);
+      if (!peel_only) {
+        // We are going to add pre-loop and post-loop (PreMainPost).
+        // But should we also multi-version for auto-vectorization speculative
```

Comment:
Just a side note: It seems that we sometimes say "multiversion" and sometimes "multi-version". I find the latter more readable. Maybe we can make it consistent at some point. This could also mean that we have `*_multi_version_*` in method names instead of `*_multiversion*_`.

Reply:
Hmm... good catch. Personally, I prefer "multiversion" as a single word. But this is very subjective :sweat_smile:

Reply:
I now renamed the 2 occurances of `multi-version` to `multiversion`, since that was a minimal diff. We could still file an RFE to rename everything later.

Reply:
Your call on what you like better since you introduced the concept :-) It should just be consistent.